org.adempiere.base/src/org/compiere/model/MRole.java

```diff
@@ -3448,4 +3448,29 @@ public MRole markImmutable() {
 		return this;
 	}
 
+	/**
+	 * Check record access through {@link #addAccessSQL(String, String, boolean, boolean)} using<br/>
+	 * either record id or record uuid
+	 * @param table
+	 * @param recordId ignore if uuid is use
+	 * @param uuid null to use recordId
+	 * @param rw true for writable, false for readonly
+	 * @return true if role has access to record
+	 */
+	public boolean checkAccessSQL(MTable table, int recordId, String uuid, boolean rw) {
+		StringBuilder sql = new StringBuilder("SELECT 1 FROM ")
+				.append(table.getTableName())
+				.append(" WHERE ")
+				.append(table.getTableName())
+				.append(".");
+		if (!Util.isEmpty(uuid, true) ) {
+			sql.append(table.getUUIDColumnName())
+				.append("=?");
+			return DB.getSQLValueEx(null, addAccessSQL(sql.toString(), table.getTableName(), true, rw), uuid) == 1;
+		} else {
+			sql.append(table.getKeyColumns()[0])
+				.append("=?");
+			return DB.getSQLValueEx(null, addAccessSQL(sql.toString(), table.getTableName(), true, rw), recordId) == 1;
```

Comment:
table.getUUIDColumnName call above is wrong (it return ad_table_uu) and I've fix it in my PR.

Reply:
> table.getUUIDColumnName call above is wrong (it return ad_table_uu) and I've fix it in my PR.

Thanks @hengsin - this fix is for mainly for release-11 as your other pull request refactor it.